lunarsky/time.py

```diff
@@ -49,7 +50,10 @@ def __init__(
         )
 
         if isinstance(location, MoonLocation):
-            self.location = location
+            if (version.major == 6 and version.minor >= 1) or version.major > 6:
+                self._location = location
+            else:
+                self.location = location
```

Comment:
Can you add a comment along the lines of what you wrote in the PR comment explaining what you are doing here and why for the future reader of the code?